homeassistant/components/ialarm/alarm_control_panel.py

```diff
@@ -0,0 +1,95 @@
+"""Interfaces with iAlarm control panels."""
+import logging
+
+from homeassistant.components.alarm_control_panel import (
+    FORMAT_NUMBER,
+    AlarmControlPanelEntity,
+)
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+)
+from homeassistant.const import CONF_PIN
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DATA_COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities) -> None:
+    """Set up a iAlarm alarm control panel based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
+    pin = entry.data.get(CONF_PIN)
+    async_add_entities([IAlarmPanel(coordinator, pin)], False)
+
+
+class IAlarmPanel(AlarmControlPanelEntity, CoordinatorEntity):
+    """Representation of an iAlarm device."""
+
+    def __init__(self, coordinator, code):
+        """Init the iAlarm device."""
+        super().__init__(coordinator)
+        self._unique_name = f"ialarm_{coordinator.host}"
```

Comment:
This unique id does not meet the unique id requirements

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

If the device doesn't have a serial number or mac you can access use `config_entry.entry_id` instead of the host

Reply:
I changed the ialarm library to expose the device mac too. Now I use it as the unique name.